runtime/decimal.c

```diff
@@ -167,6 +167,20 @@ double _bal_decimal_to_float(TaggedPtr tp) {
     return strtod(dblStr, NULL);
 }
 
+TaggedPtr _bal_decimal_from_int(int64_t val) {
+    decQuad d;
+    if (val <= INT32_MAX && val >= INT32_MIN) {
+        decQuadFromInt32(&d, val);
+        return createDecimal(&d);
+    }
+    char intStr[21]; // max length + 1
```

Comment:
How do you know 21 is enough? Use an expression that shows it.

Reply:
Added a comment